### PR TITLE
Added binary indexed tree

### PR DESCRIPTION
**Pull request by shumon84**
I added binary indexed tree for c++

### CODE REVIEW

File: code/data_structures/segment_tree/binary_indexed_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#define MAX_N 100000
+int bit[MAX_N+1];
+int n;
+
```
Comment (AdiChat) [inline code/data_structures/segment_tree/binary_indexed_tree.cpp]: You need to define the structure of the node in the form of a class 👍

**Comment by AdiChat** [comment]
The code is not correct. Moreover, you can learn about _binary index tree_ also know as _fenwick tree_ through this [C++ implementation](https://github.com/OpenGenus/cosmos/blob/master/code/data_structures/fenwick_tree/fenwick_tree.cpp)

**Comment by shumon84** [comment]
Sorry, I did not know that the Fenwick tree and the binary index tree are the same.

**Comment by AdiChat** [comment]
No problem @shumon84 Glad you learned something new 😄

**Pull request closed without merging**